### PR TITLE
#154715731 integrate paystack for payment

### PR DESCRIPTION
**Pull request by RachelAbaniwo**
#### What does this PR do?
- integrate paystack as mode of payment

#### Description of Task to be completed?
- add payments-paystack file
- add paystack as payment method to register.js file
- update index.js file in payment-paystack folder with paystack as method of payment
- create paystack.html file
- create paystack.js file
- create paystack settings form
- update paystackSettingsFormContainer.js with paystack
- create paystackapi.js file
- define paystack-pop in .eslintrc file
- add paystack script to index.html file
- update paystack.html file to include only name and email fields
- include public key and secret key instead of api key in paystackSettingsForm.js
- include public key and secret key instead of api key in paystackSettingsFormContainer.js
- add paystack header function to paystack.js file
- update paystack schema to include name and email fields
- update paystack schema to public and secret keys
- add policy.js file
- update paystackapi.js to include card input fields of name and email address instead of card details

#### How should this be manually tested?
- navigate to the product detail page, choose a variant of the product, add the product to cart, fill in 
  delivery details. 
- the paystack payment form pops up and transaction can be completed

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#154715731

#### Screenshots (if appropriate)

screen shot 2018-02-21 at 11 54 45 pm

screen shot 2018-02-21 at 11 55 04 pm

screen shot 2018-02-21 at 11 55 11 pm

screen shot 2018-02-21 at 11 57 09 pm

#### Questions:
N/A

### CODE REVIEW

File: imports/plugins/included/payments-paystack/client/settings/containers/paystackSettingsFormContainer.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    this.saveUpdate = this.saveUpdate.bind(this);
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  handleSubmit(settings) {
+    // e.preventDefault();
```
Comment (michaelogala) [inline imports/plugins/included/payments-paystack/client/settings/containers/paystackSettingsFormContainer.js]: Remove commented code.

File: imports/plugins/included/payments-paystack/server/methods/paystack-payment-methods.app-test.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      name: "Test User",
+      email: "test@test.com"
+    };
+
+    const paymentData = {
+      currency: "USD",
+      total: "19.99"
+    };
+
+    // Notice how you need to wrap this call in another function
```
Comment (michaelogala) [inline imports/plugins/included/payments-paystack/server/methods/paystack-payment-methods.app-test.js]: This comment does not describe what is happening here. Please make this comment more descriptive or remove.

File: imports/plugins/included/payments-paystack/client/checkout/paystack.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  template.$("#btn-complete-order").text(buttonText);
+  return template.$("#btn-processing").addClass("hidden");
+}
+
+function paymentAlert(errorMessage) {
+  return $(".alert").removeClass("hidden").text(errorMessage);
+}
+
+function handlePaystackSubmitError(error) {
+  const serverError = error !== null ? error.message : void 0;
```
Comment (michaelogala) [inline imports/plugins/included/payments-paystack/client/checkout/paystack.js]: What does `void 0` mean here?

**Review by michaelogala — CHANGES_REQUESTED** [review]

File: imports/plugins/included/payments-paystack/client/checkout/paystack.js (hunk 3 above)
Comment (RachelAbaniwo) [reply imports/plugins/included/payments-paystack/client/checkout/paystack.js]: ```void 0``` returns undefined.

Comment (andela-mogala) [reply imports/plugins/included/payments-paystack/client/checkout/paystack.js]: So just replace it with `undefined`

Comment (michaelogala) [reply imports/plugins/included/payments-paystack/client/checkout/paystack.js]: Replace it with `undefined` then

**Review by michaelogala — CHANGES_REQUESTED** [review]

**Review by michaelogala — DISMISSED** [review]

**Review by michaelogala — APPROVED** [review]

**Pull request merged**